### PR TITLE
Limit an organizations displayed and add an overflow count 

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
In this PR, I limit the organizations displayed on the UI and then display an overflow count.

What still needs to be done: 
- Need to check for accessibility
- It needs to be styled appropriately
- Cypress tests need to be written for its final form

## Related Tickets & Documents

- Related Issue #https://github.com/forem/forem/issues/16911

## QA Instructions, Screenshots, Recordings

A brute force method to setup your db and display these changes: 
The end goal is to create a bunch of organizations and add a user to those organizations. We should show the first 3 organization images and then a count of the overflow. 

- Create 6 more organizations by running this script
```
6.times do
    Organization.create!(
      name: Faker::Company.name,
      summary: Faker::Company.bs,
      profile_image: logo = File.open(Rails.root.join("app/assets/images/#{rand(1..40)}.png")),
      nav_image: logo,
      url: Faker::Internet.url,
      slug: "org#{rand(10_000)}",
      github_username: "org#{rand(10_000)}",
      twitter_username: "org#{rand(10_000)}",
      bg_color_hex: Faker::Color.hex_color,
      text_color_hex: Faker::Color.hex_color,
    )
  end
```
- Delete all organizations for the first user 
```
User.first.organizations.delete_all
```

- Thereafter seed the fist user with all the organizations that you have in your db: 
```
Organization.all.each do |org|
  OrganizationMembership.create!(user_id: User.first.id, organization_id: org.id, type_of_user: "admin")
end
```
- Now look for the first user and you should see the first three organizations + an overflow count.

An example screenshot of a user with organizations: 

Show the first 3 organizations with an overflow of 10 organizations:
screencapture-localhost-3000-admin-users-2022-04-19-17_17_38

With tooltip:
Screenshot 2022-04-19 at 17 17 53

### UI accessibility concerns?

It will be tackled in a subsequent PR 

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/helpers/admin/users_helper.rb [hunk 1]
```diff
@@ -45,5 +45,37 @@ def user_status(user)
...
+    #  @param [Integer] The limit of organizations that we show
+    # @return [String]
+    def organization_tooltip(organization_names, count, imposed_limit: 0)
+      str = organization_names.first(imposed_limit).join(", ").to_s
+
+      return str unless count > imposed_limit
+
+      overflow = overflow_count(count, imposed_limit: imposed_limit)
+      if overflow == 1
+        str + " & #{overflow_count(count, imposed_limit: imposed_limit)} other."
```
Comment (aitchiss) [inline app/helpers/admin/users_helper.rb:75]: It's a tiny thing but I don't think we want the full stop (it's not in the design, and I feel like it looks a bit weird when it's not a full sentence 🤔  I don't mind if this gets swept up in the later UI update though 

```suggestion
        str + " & #{overflow_count(count, imposed_limit: imposed_limit)} other"
```

File: spec/helpers/admin/users_helper_spec.rb [hunk 2]
```diff
@@ -118,4 +118,42 @@
...
+        expect(tooltip).to eq "org1, org2 & 2 others."
+      end
+    end
+
+    context "when the limit is more than the total" do
+      it "renders the correct tooltip" do
+        tooltip = helper.organization_tooltip(%w[org1], 1, imposed_limit: 2)
+        expect(tooltip).to eq "org1"
+      end
+    end
```
Comment (aitchiss) [inline spec/helpers/admin/users_helper_spec.rb:157]:
```suggestion
  describe "#organization_tooltip" do
    context "when the limit is less than the total" do
      it "renders the correct tooltip when the array of items do not match the imposed limit" do
        tooltip = helper.organization_tooltip(%w[org1 org2 org3], 3, imposed_limit: 2)
        expect(tooltip).to eq "org1, org2 & 1 other"
      end

      it "renders the correct tooltip for an overflow of 1" do
        tooltip = helper.organization_tooltip(%w[org1 org2], 3, imposed_limit: 2)
        expect(tooltip).to eq "org1, org2 & 1 other"
      end

      it "renders the correct tooltip for an overflow of more than 1" do
        tooltip = helper.organization_tooltip(%w[org1 org2], 4, imposed_limit: 2)
        expect(tooltip).to eq "org1, org2 & 2 others"
      end
    end

    context "when the limit is more than the total" do
      it "renders the correct tooltip" do
        tooltip = helper.organization_tooltip(%w[org1], 1, imposed_limit: 2)
        expect(tooltip).to eq "org1"
      end
    end
```

Just adding suggestion to remove the full stops

**Review by aitchiss — APPROVED** [review]
Works great! I've left some minor suggestions but nothing blocking

**Review by Zhao-Andy — APPROVED** [review]
LGTM! Note that I had to turn on the member view feature flag 😅

**Comment by Ridhwana** [comment]
> LGTM! Note that I had to turn on the member view feature flag 😅

Oops sorry @Zhao-Andy I forgot to mention that 🤦🏽‍♀️

**Pull request merged**